### ISSUE TITLE
What should F5 do?

### ISSUE THREAD

**Issue by MrAmericanMike** [microsoft/vscode-generator-code#369]
So I used `yo code` to generate a template for a Snippets Extension.

Reading `vsc-extension-quickstart.md` I see that it mentions ```Press `F5` to open a new window with your extension loaded.```

However when I press F5 I get an error saying:

```
You don't have an extension for debugging "JSON with Comments". Should we find a "JSON with Comments" extension in the Marketplace?
```

Which I assume is not what should happen and I guess it's because my F5 key is doing something different.

What would be the way to get this working in order to test my Extension not having to manually copy it into `.vscode/extensions` manually each time?

**Comment by aeschli** [comment]
I'm not able to reproduce. Fro me F5 runs the provided launch config.

@isidorn Do you have some insights how the message above could have been shown?

**Comment by isidorn** [comment]
Yes it could happen that the user did not have a `launch.json` or that via a bug debugging was not aware of configuration in the launch.json. And then pressing F5 we would try to debug the current file by asking the debug extension to generate a launch configuration dynamically. Here we tried to do that for the JSON with Comments file.

fyi @roblourens

**Comment by roblourens** [comment]
I tried it too and it worked. @MrAmericanMike do you have a `.vscode/launch.json` file? Please make sure you haven't deleted that by accident.

**Comment by sharktide** [comment]
I had the same issue and I think I found out the problem. 

![Image](https://github.com/user-attachments/assets/69d5b7aa-49d0-4686-92c0-dcf5467036b7)

In the bottom right corner, visual studio code may categorize it as "JSON with comments" Clicking on it and changing it to plain JSON worked for me after removing any json comments from the file.

**Comment by sharktide** [comment]
If it is helpful I recreated the problem (I had this issue too) and solved it. 

1. VScode has to be open to the folder where the .vscode/launch.json is. If vscode is open to the parent folder, it will not work. I suspect what @MrAmericanMike did was initialize a git repository and then run the yeoman code generator so it would generate a subfolder inside the parent directory that VScode was open in, and since the .vscode/launch.json was not in the folder that vscode was open in, it did not launch correctly.

**Comment by MrAmericanMike** [comment]
> If it is helpful I recreated the problem (I had this issue too) and solved it.
> 
> 1. VScode has to be open to the folder where the .vscode/launch.json is. If vscode is open to the parent folder, it will not work. I suspect what [@MrAmericanMike](https://github.com/MrAmericanMike) did was initialize a git repository and then run the yeoman code generator so it would generate a subfolder inside the parent directory that VScode was open in, and since the .vscode/launch.json was not in the folder that vscode was open in, it did not launch correctly.

Yes I usually work on a parent folder in VSCode and not opening the one of the project itself.

So for example `VSCode/Extensions/some_extension` I probably have a workspace under `Extensions` and that's what I open, and not directly `some_extension` So if VSCode is looking for a .vscode folder on the root of what I have opened, it wont find it.
